components/service/pocket/src/main/java/mozilla/components/service/pocket/ext/Mappers.kt

```diff
@@ -43,3 +47,31 @@ internal fun PocketStoryEntity.toPocketRecommendedStory(): PocketRecommendedStor
  */
 internal fun PocketRecommendedStory.toPartialTimeShownUpdate(): PocketLocalStoryTimesShown =
     PocketLocalStoryTimesShown(url, timesShown)
+
+/**
+ * Map sponsored Pocket stories to the object type that we persist locally.
+ */
+internal fun ApiSpoc.toLocalSpoc(): SpocEntity =
+    SpocEntity(
+        url = url,
+        title = title,
+        imageUrl = imageSrc,
+        sponsor = sponsor,
+        clickShim = shim.click,
+        impressionShim = shim.impression
```

Comment:
I see that we're persisting these two shims, but not where we're reading and using them? Can you clarify?

Reply:
They are added back in a new `PocketSponsoredStoryShim` data class by `toPocketSponsoredStory()` used [when `getAllSpocs()` is called.](https://github.com/mozilla-mobile/android-components/blob/13682c5b1d51d1ebcbcdc5e685f1c68b259443fa/components/service/pocket/src/main/java/mozilla/components/service/pocket/spocs/SpocsRepository.kt#L27-L29)

Reply:
Yes, OK so that's where you read them back from storage, but I don't see anything using these values, other than tests?

Reply:
They will be used in Fenix's telemetry similar to desktop - https://bugzilla.mozilla.org/show_bug.cgi?id=1550119

Reply:
I see, OK. I think we will need some planning / discussions around how this should work, but that's separate from this PR. :)